lib/QueryRunner.js

```diff
@@ -47,16 +47,34 @@ export default class QueryRunner {
 
     if (userOptions.port) {
       if (!userOptions.givenPortOnly) {
-        if (gameQueryPortOffset) { addAttemptWithPort(userOptions.port + gameQueryPortOffset) }
+
+        if (gameQueryPortOffset) {
+          if (Array.isArray(gameQueryPortOffset)) {
```

Comment:
The repeating if's could be resolved be preprocessing `gameQueryPortOffset`.
Before the first if, check if it is an array, if not, make one by the sole value (if any).

Afterwards, its just a matter of `.forEach` it and also the `if (gameQueryPortOffset)` will not be required anymore.

Reply:
I think I understood what you meant. Thank you! 🙌 
Just added a new code. Let me know!